src/beautifiers/index.coffee

```diff
@@ -241,15 +240,19 @@ module.exports = class Beautifiers extends EventEmitter
         logger.info('beautify', text, allOptions, grammar, filePath, onSave)
         logger.verbose(allOptions)
 
-        # Get language
-        fileExtension = path.extname(filePath)
-        # Remove prefix "." (period) in fileExtension
-        fileExtension = fileExtension.substr(1)
-        languages = @languages.getLanguages({grammar, extension: fileExtension})
-        logger.verbose(languages, grammar, fileExtension)
+        languages = []
+        fileExtension = null
+        getLanguages = (filePath) =>
+          # Get language
+          fileExtension = path.extname(filePath)
+          # Remove prefix "." (period) in fileExtension
+          fileExtension = fileExtension.substr(1)
+          languages = @languages.getLanguages({grammar, extension: fileExtension})
+          logger.verbose(languages, grammar, fileExtension)
+          return languages
 
         # Check if unsupported language
-        if languages.length < 1
+        if !filePath || getLanguages(filePath).length < 1
```

Comment:
This is not true. In the case where a `filePath` is `undefined` (i.e. falsy) we may still find a language for the grammar provided by Atom.
Simply removing the `!filePath` conditional should be sufficient.